src/bun.js/api/bun/socket.zig

```diff
@@ -814,18 +814,40 @@ pub const Listener = struct {
         socket.setTimeout(120000);
     }
 
-    // pub fn addServerName(this: *Listener, _: *JSC.JSGlobalObject, _: *JSC.CallFrame) callconv(.C) JSValue {
-
-    //     uws.us_socket_context_add_server_name
-    // }
+    pub fn addServerName(this: *Listener, global: *JSC.JSGlobalObject, hostname: JSValue, tls: JSValue) JSValue {
+        if (!this.ssl) {
+            global.throwInvalidArguments("addServerName requires SSL support", .{});
+            return .zero;
+        }
+        if (!hostname.isString()) {
+            global.throwInvalidArguments("hostname pattern expects a string", .{});
+            return .zero;
+        }
+        const host_str = hostname.toSliceZ(
```

Comment:
The poorly-named `toSliceZ`does not actually make it `[:0]const u8`

https://github.com/oven-sh/bun/blob/c5270f812105133975f343c4618c2411f5d899f3/src/bun.js/bindings/bindings.zig#L804-L821